src/components/JoinList.js

```diff
@@ -0,0 +1,49 @@
+import { getDoc } from 'firebase/firestore';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { getListFromDB } from '../lib/api';
+
+export const JoinList = () => {
+  const [joinToken, setJoinToken] = useState('');
+  const navigate = useNavigate();
+
+  const checkIfTokenExist = async (token) => {
+    const docRef = getListFromDB(token);
+    const docSnap = await getDoc(docRef);
+
+    if (docSnap.exists()) {
+      localStorage.setItem('token', joinToken);
+      navigate('/list');
+    } else {
+      alert('No existe este token!');
+    }
+  };
+
+  const handleChange = (e) => setJoinToken(e.target.value);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    checkIfTokenExist(joinToken);
+  };
+
+  return (
+    <>
+      <h3>Join list</h3>
+      <form onSubmit={handleSubmit}>
+        <p>Join an existing shopping list by entering a three word token.</p>
+        <label htmlFor="joinToken">Share token</label>
+        <br />
+        <input
+          id="joinToken"
+          type="text"
+          value={joinToken}
+          onChange={handleChange}
+          placeholder="Enter your token"
+        ></input>
```

Comment:
- No es necesario `</input>`.
- Se produce un error, cuando el token es una cadena vacía.
- Sería interesante, dar un poco de padding al input `padding: 5px` esta "muy estrecho" (en los estilos)
![image](https://user-images.githubusercontent.com/8689897/151667672-fe042e2b-fad3-496d-970d-4daa75056a89.png)


```suggestion
        <input
          id="joinToken"
          type="text"
          value={joinToken}
          onChange={handleChange}
          placeholder="Enter your token"
          required
        />
```

Reply:
Solucionado :)